docs/developer/architecture/authentication.rst

```diff
@@ -0,0 +1,97 @@
+Authentication Architecture Document
+====================================
+
+Context:
+--------
+Currently, when we deploy MetalK8s we pre provision super admin user with admin
+password. Meaning anyone wanting to use K8S/Salt APIs should authenticate using
+this single super admin user. Another way to access the APIs is by using K8S
+admin certificate that is stored in /etc/kubernetes/admin.conf. We could also
+manually provisioned other users and credentials as well as roles binding
+but the way the functionality is accessible currently is too complex and not
+enough secured since user / password are stored in clear text.
+This is not good enough in term of security and flexibility.
+
+We would like at least to be able to add different users with different
+credentials and ideally to integrate K8S authentication system with external
+identity provider.
+
+Managing K8S role binding between user / groups High level roles and K8S roles
+is not part of this specification.
+
+Requirements:
+-------------
+
+Basically we are talking about:
+
+- Being able to provision users with an embedded Identity provider (IDP)
+- Being able to integrate with an external IDP
+
+`Dex <https://github.com/dexidp/dex>`_ seems to be the preferred solution for
+now, for both local identity provider or proxy to external IDP.
+Integration with LDAP or Microsoft AD are the most important ones to support.
+
+User Stories:
+-------------
+**User Management with Dex as an IDP**
+
+As an IT Generalist, I want to provision/edit usersa and high level roles.
+group provisioning not supported by Dex from what I understand.
+The Product high level roles are:
+
+- Cluster Admin role
+- Solution Admin role
+- Read Only
+
+This is done from CLI with well documented procedure.
+Entered passwords are never visible and encrypted when stored in DEX DB.
+The CLI tool enables to add / delete and edit passwords and roles.
+
+The CLI tool is used right after bootstrap node deployment to provision the
+very first user (which won't be provisioned anymore by default)
```

Comment:
There's another req that should be added, IMHO, about the Vagrant deployment.

Similar to how we create (sparse) volumes for Prometheus and such by default when running `vagrant up`, I think it could make sense to also pre-provision some well-known user in said environment to retain a cluster that's fully working out of the box.

Reply:
Actually I changed this req and put back the need of having a pre-provisioned user so I guess it would not be different with vagrant N